operator/archivecontroller/executor.go

```diff
@@ -67,19 +86,20 @@ func (a *ArchiveExecutor) jobName() string {
 	return k8upv1.ArchiveType.String() + "-" + a.Obj.GetName()
 }
 
-func (a *ArchiveExecutor) setupArgs(archive *k8upv1.Archive) []string {
-	args := []string{"-archive", "-restoreType", "s3"}
-
-	if archive.Spec.RestoreSpec != nil {
-		if len(archive.Spec.RestoreSpec.Tags) > 0 {
-			args = append(args, executor.BuildTagArgs(archive.Spec.RestoreSpec.Tags)...)
-		}
+func (a *ArchiveExecutor) setupArgs() ([]string, error) {
```

Comment:
The only return in this function returns a nil error. So it doesn't really need to return an error here.

Reply:
Actually I am using code style as like https://github.com/k8up-io/k8up/blob/1088118931d95579d76d4b0b668214f21d63d01d/operator/restorecontroller/executor.go#L90, maybe in feature we add more options with a use case and in this use case we check something generate a error.

In this case no eed change function definition and just add our condition.

Reply:
If we'd ever add something more in the future that might return an error, we can always adjust the function signature then.

I'd like to avoid having non-functional code lines, as it will only confuse people.

Reply:
Okay, You're right
I will change it

Reply:
fixed